interfaces/builtin/fwupd.go

```diff
@@ -64,13 +68,78 @@ const fwupdPermanentSlotAppArmor = `
   /sys/firmware/efi/esrt/entries/** r,
 
   # Allow fwupd to access system information
-  /sys/devices/virtual/dmi/id/product_name r,
-  /sys/devices/virtual/dmi/id/sys_vendor r,
+  /sys/firmware/dmi/tables/smbios_entry_point r,
+  /sys/devices/virtual/dmi/id/{sys,bios,product,chassis,board}_{name,version,vendor,date,serial,sku,type,asset_tag,family} r,
+
+  /sys/firmware/dmi/tables/DMI r,
+  /sys/bus/usb/devices/ r,
+
+  /sys/firmware/acpi/bgrt/{type,version,{x,y}offset,image} r,
 
   # Allow read/write access for efivarfs filesystem
   /sys/firmware/efi/efivars/ r,
   /sys/firmware/efi/efivars/** rw,
 
+  /proc/modules r,
+  /run/udev/data/* r,
+  /sys/devices/**/usb*/busnum r,
+  /sys/devices/**/usb*/**/busnum r,
+  /sys/devices/**/usb*/devnum r,
+  /sys/devices/**/usb*/**/devnum r,
+  /sys/devices/**/usb*/speed r,
+  /sys/devices/**/usb*/**/speed r,
+  /sys/devices/**/usb*/descriptors r,
+  /sys/devices/**/usb*/**/descriptors r,
+  /sys/devices/**/usb*/bConfigurationValue r,
+  /sys/devices/**/usb*/**/bConfigurationValue r,
+  /dev/mem r,
+  /sys/kernel/security/lockdown r,
+  owner @{PROC}/@{pid}/mountinfo r,
+  /sys/bus/pci/devices/ r,
+  /sys/devices/pci*/**/vendor r,
+  /sys/devices/pci*/**/device r,
+  /sys/devices/pci*/**/revision r,
+  /sys/devices/pci*/**/subsystem_vendor r,
+  /sys/devices/pci*/**/subsystem_device r,
+  /sys/devices/pci*/**/class r,
+  /sys/devices/pci*/**/config r,
+  /sys/devices/pci*/**/name r,
+  /sys/devices/**/serio*/description r,
+  /dev/cpu/*/msr rw,
+
+  /sys/devices/**/description r,
+
+  /sys/devices/**/mei/*/fw_ver r,
+  /sys/devices/**/unique_id r,
+
+  /sys/devices/**/nvm_version r,
+  /sys/devices/**/generation r,
+  /sys/devices/**/vendor_name r,
+  /sys/devices/**/nvmem r,
+  /sys/devices/**/authorized r,
+
+  capability sys_rawio,
+  capability sys_nice,
+
+  owner @{PROC}/@{pid}/mounts r,
+  /run/mount/utab r,
+  /sys/bus/*/devices/ r,
+  /sys/devices/**/i2c*/**/name r,
```

Comment:
need `/dev/i2c-*` for the Realtek MST plugin

Reply:
It's design dependent and also kernel dependent.  It's whatever I2C channel has been allocated for the DP communication on that design.  If the kernel is configured to not load some of the other I2C buses, the number may jump around.